pysindy/pysindy.py

```diff
@@ -141,10 +142,16 @@ def fit(self, x, t=1, x_dot=None, multiple_trajectories=False):
             for each trajectory. If False, the training data must be a single
             array.
 
+        quiet: boolean, optional (default False)
+            Whether or not to suppress warnings during model fitting.
+
         Returns
         -------
         self: returns an instance of self
         """
+        if quiet:
+            warnings.filterwarnings("ignore")
```

Comment:
We should be more precice here and only filter the appropriate warnings in the scope of the fit function:

```python
action = "ignore" if quiet else "default"
with warnings.catch_warnings(): 
    warnings.filterwarnings(action, category=ConvergenceWarning)
    pass # fit code here
```

Reply:
I was also thinking users might want to filter out other warnings that can be raised during fitting like LinAlgWarning (raised by STLSQ when the matrix is ill-conditioned). I suppose it's clearer if we explicitly state which warnings we are aiming to ignore. I'll update the code now.